### PR TITLE
sql: add telemetry for schema change usage commands

### LINKED ISSUE #44316: sql: Telemetry needed for Schema Changes Usage
**Is your feature request related to a problem? Please describe.**
Towards our assessing the Operator/Database Performance Engineer experience, it appears we're unable to query schema changes usage. 

**Describe the solution you'd like**
I'd like to be able to query [schema changes](https://www.cockroachlabs.com/docs/stable/online-schema-changes.html) by external users in the Features table in the Registration Cluster.

**Describe alternatives you've considered**
I've attempted to query the Features table with "CREATE," "create," "DROP," "drop," "ALTER," and "alter." The results are not clear.

### PR DESCRIPTION
**Pull request by otan**
Resolves https://github.com/cockroachdb/cockroach/issues/44316.

This PR adds telemetry for usage of various CREATE, ALTER and DROP
commands.

Telemetry names may be subcategorized by different types, e.g.:

```
sql.schema.alter_table
sql.schema.alter_table.add_check_constraint
sql.schema.alter_table.add_column
sql.schema.alter_table.add_constraint
sql.schema.alter_table.column_mutation
sql.schema.alter_table.drop_column
sql.schema.alter_table.drop_constraint
sql.schema.alter_table.set_audit
sql.schema.alter_table.validate_constraint
```

Release note: None

### CODE REVIEW

File: pkg/sql/alter_table.go [hunk 1]
```diff
@@ -324,7 +330,8 @@ func (n *alterTableNode) startExec(params runParams) error {
 					"session variable experimental_enable_primary_key_changes is set to false, cannot perform primary key change")
 			}
 
-			// Increment telemetry about uses of primary key changes.
+			// Increment telemetry about supported uses of primary key changes
```
Comment (rohany) [inline pkg/sql/alter_table.go]: I feel like you should just delete this counter in favor of the new one you added

**Comment by rohany** [comment]
I wonder if there is a way to add a getTelemtryCounter method on the `*tree.AlterTableCmd` node so that we could just call that at the top of the loop rather than have to do it in each case.

File: pkg/sql/alter_table.go (hunk 1 above)
Comment (otan) [reply pkg/sql/alter_table.go]: sure thing!

**Comment by otan** [comment]
> I wonder if there is a way to add a getTelemtryCounter method on the *tree.AlterTableCmd node so that we could just call that at the top of the loop rather than have to do it in each case.

 how do you mean?

**Comment by rohany** [comment]
The `AlterTable` node has a list of `AlterTableCmds`. I'm thinking we could make an interface like `AlterTableCmdWithTelemetry` that has a method `GetTelemetryCounter`. This would be implemented for the cases that you wrote up here, and then at the top of the loop we can just check if the command is an `AlterTableCmdWithTelemetry` and get the counter. Or better yet, enforce that we have telemetry about all types that ascribe to the `AlterTableCmd` interface.

**Comment by otan** [comment]
Done.

**Review by rohany — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44766#-:-M-LfPX7D4ieax8ZYPqd:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rohany)

**Comment by otan** [comment]
bors r=rohany

**Comment by otan** [comment]
bors r=rohany

**Comment by otan** [comment]
hitting flakes and rebase conflicts :(

bors r=rohany

**Comment by otan** [comment]
flake again

bors r=rohany

**Comment by rohany** [comment]
5th times the charm

**Comment by otan** [comment]
pulled a homer

**Pull request merged**